packages/react-composites/tests/app/callwithchat/HermeticApp.tsx

```diff
@@ -11,15 +17,48 @@ import { QueryArgs } from './QueryArgs';
  *
  */
 export function HermeticApp(props: { queryArgs: QueryArgs }): JSX.Element {
-  throw new Error(`Unimplemented, though you tried to render ${JSON.stringify(props.queryArgs)}`);
-  return <></>;
+  const { queryArgs } = props;
+  if (!queryArgs.fakeChatAdapterArgs) {
+    return (
+      <h3>
+        ERROR: <code>fakeChatAdapterArgs</code> is unset.
+      </h3>
+    );
+  }
+  if (!queryArgs.mockCallAdapterState) {
+    return (
+      <h3>
+        ERROR: <code>mockCallAdapterState</code> is unset.
+      </h3>
+    );
+  }
+  return <HermeticAppImpl queryArgs={queryArgs as HermeticQueryArgs & CommonQueryArgs} />;
+}
+
+function HermeticAppImpl(props: { queryArgs: HermeticQueryArgs & CommonQueryArgs }): JSX.Element {
+  const { queryArgs: args } = props;
+
+  const [callAdapter, setCallAdapter] = useState<CallAdapter | undefined>(undefined);
+  useEffect(() => {
+    (async (): Promise<void> => {
+      console.log('Creating mock adapter with args', args.mockCallAdapterState);
+      setCallAdapter(new MockCallAdapter(args.mockCallAdapterState));
```

Comment:
Correct me if I am mistaken, but I don't see an arg for alternateCallerId. I suppose for now the HermeticApp will not work with PSTN calls?

Reply:
As long as `alternateCallerId` is in the adapter state, tests can initialize it to required value.

In fact, there are already hermetic tests for CallComposite that set `alternateCallerId`.